Samples/Win32-Http/main.cpp

```diff
@@ -263,24 +279,27 @@ void DoHttpCall(std::string url, std::string requestBody, bool isJson, std::stri
 
         HCHttpCallResponseGetNetworkErrorCode(call, &networkErrorCode, &platErrCode);
         HCHttpCallResponseGetStatusCode(call, &statusCode);
-        HCHttpCallResponseGetResponseString(call, &str);
-        if (str != nullptr) responseString = str;
-        std::vector<std::vector<std::string>> headers = ExtractAllHeaders(call);
-
-        if (!isJson)
+        if (!customWriteUsed)
         {
-            size_t bufferSize = 0;
-            HCHttpCallResponseGetResponseBodyBytesSize(call, &bufferSize);
-            uint8_t* buffer = new uint8_t[bufferSize];
-            size_t bufferUsed = 0;
-            HCHttpCallResponseGetResponseBodyBytes(call, bufferSize, buffer, &bufferUsed);
-            HANDLE hFile = CreateFileA(filePath.c_str(), GENERIC_WRITE, 0, NULL, CREATE_NEW, FILE_ATTRIBUTE_NORMAL, NULL);
-            DWORD bufferWritten = 0;
-            WriteFile(hFile, buffer, (DWORD)bufferUsed, &bufferWritten, NULL);
-            CloseHandle(hFile);
-            delete[] buffer;
+            HCHttpCallResponseGetResponseString(call, &str);
+            if (str != nullptr) responseString = str;
+
+            if (!isJson)
+            {
+                size_t bufferSize = 0;
+                HCHttpCallResponseGetResponseBodyBytesSize(call, &bufferSize);
+                uint8_t* buffer = new uint8_t[bufferSize];
+                size_t bufferUsed = 0;
+                HCHttpCallResponseGetResponseBodyBytes(call, bufferSize, buffer, &bufferUsed);
+                HANDLE hFile = CreateFileA(filePath.c_str(), GENERIC_WRITE, 0, NULL, CREATE_NEW, FILE_ATTRIBUTE_NORMAL, NULL);
```

Comment:
Not exactly sure what this write file bit doing, but I don't think that it happens for the customWriteUsed flow.

Reply:
The file writing should be skipped when a custom write is used since its also encapsulated in the !customWriteUsed if statement.